### DISCUSSION TITLE
Error: Cannot find package 'remark'

### DISCUSSION THREAD

**Discussion by gnought** [remarkjs/remark#1333] [Q&A] [answered]
I installed remark v3.0.0 VS code extension, however it prompts an following error. The lib/node_modules path does not exist. Do anyone have an idea how to fix it?

```
Error: Cannot find package 'remark' imported from /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Plugin).app/Contents/lib/node_modules
    at new n (/Users/builder/.vscode/extensions/unifiedjs.vscode-remark-3.0.0/out/remark-language-server.js:810:16860)
    at Ox (/Users/builder/.vscode/extensions/unifiedjs.vscode-remark-3.0.0/out/remark-language-server.js:812:7057)
    at u5 (/Users/builder/.vscode/extensions/unifiedjs.vscode-remark-3.0.0/out/remark-language-server.js:812:7414)
    at Px (/Users/builder/.vscode/extensions/unifiedjs.vscode-remark-3.0.0/out/remark-language-server.js:812:8570)
    at Wg (/Users/builder/.vscode/extensions/unifiedjs.vscode-remark-3.0.0/out/remark-language-server.js:812:8759)
    at zl (/Users/builder/.vscode/extensions/unifiedjs.vscode-remark-3.0.0/out/remark-language-server.js:812:9911)
    at Ma (/Users/builder/.vscode/extensions/unifiedjs.vscode-remark-3.0.0/out/remark-language-server.js:812:9163)
    at b (/Users/builder/.vscode/extensions/unifiedjs.vscode-remark-3.0.0/out/remark-language-server.js:831:637)
    at y (/Users/builder/.vscode/extensions/unifiedjs.vscode-remark-3.0.0/out/remark-language-server.js:831:1896)
    at async /Users/builder/.vscode/extensions/unifiedjs.vscode-remark-3.0.0/out/remark-language-server.js:831:3033
```

**Comment by ChristianMurphy** [comment]
Welcome @gnought! 👋 
Sorry you ran into a spot of confusion.

Make sure the project folder you are opening has remark-cli and the plugins you want installed in a package.json file.

> To configure this extension, you define your preferred markdown style in a configuration file (.remarkrc, .remarkrc.js, etc. or in package.json). This file is picked up by vscode-remark and other tools (useful for contributors that don’t use VS Code).
>
> The configuration file is also used by [remark-cli](https://github.com/remarkjs/remark/tree/main/packages/remark-cli), which is recommended to be used alongside vscode-remark, as an npm script and/or in CI, to enforce the markdown style.
>
> To use this extension, set up [remark-cli](https://github.com/remarkjs/remark/tree/main/packages/remark-cli) like you normally would in your project with a configuration file. The extension will find the configuration in your workspace just like running remark-cli in your terminal would. You will be able to see your linter work as you type and you can format your code if you want it to.
>
> Now, you can open markdown files in your project, and you’ll see squiggly lines and warnings in the Problems pane if the code doesn’t adhere to the coding standards.

Source: project readme: https://github.com/remarkjs/vscode-remark?tab=readme-ov-file#what-is-this

**Reply by remcohaszing** [reply]
What you really need is for `remark` to be available, not `remark-cli`. `remark` is usually available as a transient dependency of `remark-cli`. However, as of version 3 you don’t need to have `remark` installed anymore. `remark-language-server` / `vscode-remark` tries to resolve `remark`, but falls back to its own bundled version of remark.

I suspect the project where this occurs has an old version of `remark` installed.

@gnought Could you share some more context? Is your project open source? Do you have `remark` installed? If so, which version? The output `npm ls remark` / `yarn why remark` / `pnpm ls remark` / `cat node_modules/remark/package.json` would be useful.

**Reply by gnought** [reply]
Thanks @ChristianMurphy @remcohaszing inputs.

I am a newbie, trying to switch from markdownlint.

I don't have any remark program installed, just installed the vscode-remark VS extension.

Does it mean I need to install remark npm package per project?

**Reply by wooorm** [reply]
markdownlint is a batteries included vscode thing.
This vscode thing instead assumes you have remark setup in projects, and works with that.
See the readme for more info on what it is, and when to use it: https://github.com/remarkjs/vscode-remark

**Reply by remcohaszing** [reply, marked as answer]
Ideally you should have `remark` installed per project. `remark-cli` installs this as a transitive dependency. This ensures the same `remark` version is used both by the VSCode extension and when running `remark` on the command line.

However, if `remark` isn’t found, the VSCode extension falls back to a bundled version of `remark`. For me it runs fine regardless whether or not `remark` is installed in the project, so I’m surprised you’re seeing this error.
